proposals/052-cryptography-leg-1.md

```diff
@@ -0,0 +1,329 @@
+
+## Goals and Non-goals
+
+The following are the immediate goals of this first leg:
+
+- To publish the `botan`, `botan-low`, and `botan-bindings` libraries
+- To build proper documentation and tutorials covering the subject material
+- To establish a community project presence focused on Haskell and cryptography
+
+The following are overarching goals, and will not be complete by this first leg:
+
+- To publish the `botanium` libsodium- and `botanium` crypton- compatible libraries
+- To publish the `crypto-schemes` library and `crypto-schemes-botan` backend.
+- To provide a suitable alternative interface / backend (as an option) for `crypton`, `tls`, `x509`
+- To consolidate Haskell cryptography into a consistent, unified, user-friendly system
+- To enable the safe use and implementation of advanced cryptographic data structures
+- To provide cryptographic typeclasses for safe and consistent use
+
+The following are not goals:
+
+- Finishing every goal by the first leg of the project
+- Matching interfaces 1:1 with existing libraries / interfaces
+- Implementing every algorithm permutation possible
+
+## Timeline
+
+This timeline will focus on the first leg of the proposal, and the immediate goal of publishing the following libraries:
+
+- botan-bindings
+- botan-low
+- botan
+
+The full project has a large scope, and will require significant time to complete. More advanced concepts such as typeclasses may be delivered in a future leg focused on generalizing.
+
+### Expected project leg duration
+
+We estimate that it will take 3 months for this leg, to achieve our goal of developing and publishing these libraries. This estimate is based on the significant amount of work already completed, the length of time in which it was accomplished, and the remaining amount of work.
+
+Several work items (strictness, memory safety, C++ shims for missing FFI functionality) are difficult to estimate, and so time has been allocated for them in the estimate, but their delivery is considered optional, and may roll over into a future leg as necessary.
+
+### Intermediate deliverables
+
+The following are intermediate deliverables intended to keep the community apprised of any progress that is made, and any challenges that arise, and to seek community feedback:
+
+- Frequent (2-3x weekly) activity update on devlog
+- Weekly major update that includes github push
+- Monthly report that summarizes what specific items have been worked on, what items will be worked on, and what challenges have arisen, if any.
+
+### Deliverables
+
+The following are the intended deliverables of this leg of the proposal:
+
+- `botan-bindings`
+- `botan-low`
+- `botan`
+
+Each library should be beta- or / production-ready, and submitted to hackage as a candidate or accepted package. 
+
+Each library should have:
+
+- Documentation
+- Tutorials and examples
+- Tracked issues
+- Unit tests
+
+A project website will also be created.
```

Comment:
Why? What value does it bring beyond that in the Github repo?

Reply:
The idea was that it is low-effort and aids in discoverability. Many successful Haskell libraries have at least a minor web presence, and it could be as simple as a page or site hosted on github.io linked to from the haskell cryptography group page. It would also be a place to link to funding / donations mechanisms and to acknowledge supporters and contributors.

This can be moved to optional deliverables / nice-to-haves, or elided completely.